### PR TITLE
MULE-13290: Extension model is missing types and extensions declare own any

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: modules/extensions-spring-support/src/test/resources/models/marvel.json [hunk 1]
```diff
@@ -3560,23 +3556,11 @@
...
       "type": "CUSTOM_ERROR",
       "namespace": "MARVEL",
       "parent": {
         "type": "ANY",
-        "namespace": "MARVEL",
-        "parent": {
-          "type": "ANY",
-          "namespace": "MULE"
-        }
+        "namespace": "MULE"
```
Comment (alegmarra) [inline modules/extensions-spring-support/src/test/resources/models/marvel.json]: why are we losing the MARVEL:ANY here?

File: core/src/main/java/org/mule/runtime/core/api/exception/Errors.java [hunk 2]
```diff
@@ -151,7 +149,14 @@
     public static final String SOURCE_ERROR_RESPONSE_GENERATE_ERROR_IDENTIFIER = "SOURCE_ERROR_RESPONSE_GENERATE";
 
     /**
-     * Indicates that a severe error occurred. Cannot be handled. Other unhandleable errors should go under it.
+     * Indicates that an unknown and unexpected error occurred. Cannot be handled directly, only through ANY.
+     */
+    public static final String UNKNOWN_ERROR_IDENTIFIER = "UNKNOWN";
+
+    // UNHANDLEABLE
```
Comment (alegmarra) [inline core/src/main/java/org/mule/runtime/core/api/exception/Errors.java]: probably better to communicate this in a more explicit way

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ErrorsModelFactory.java [hunk 3]
```diff
@@ -132,9 +133,12 @@ private String getErrorNamespace(ErrorTypeDefinition errorType) {
   private void addType(ErrorTypeDefinition<?> errorType,
                        Graph<ErrorTypeDefinition, Pair<ErrorTypeDefinition, ErrorTypeDefinition>> graph) {
     graph.addVertex(errorType);
-    ErrorTypeDefinition parentErrorType = errorType.getParent().orElse(ANY);
-    graph.addVertex(parentErrorType);
-    graph.addEdge(errorType, parentErrorType);
+    String type = errorType.getType();
+    if (!ANY.name().equals(type) && !CRITICAL.name().equals(type)) {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ErrorsModelFactory.java]: any reason not to allow ANY in other namespaces?

Comment (afelisatti) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ErrorsModelFactory.java]: It messes up with the hierarchy, connectivity and retry are mapped to Mule's types, but any others to their own ANY. This makes it consistent. If we want a way to match all errors from an extension then we can add a special "namespace" matcher in the future. Today you could use the "when" attribute and use a custom expression.

File: core/src/main/java/org/mule/runtime/core/api/exception/Errors.java (hunk 2 above)
Comment (afelisatti) [inline core/src/main/java/org/mule/runtime/core/api/exception/Errors.java]: Identifiers are rarely used alone. I'll add a distinction within the ComponentIdentifiers.

**Pull request merged**